openapi_python_client/parser/properties/__init__.py

```diff
@@ -486,8 +494,10 @@ def _property_from_data(
         )
     elif data.type == "array":
         return build_list_property(data=data, name=name, required=required, schemas=schemas, parent_name=parent_name)
-    elif data.type == "object":
+    elif data.type == "object" or data.allOf:
         return build_model_property(data=data, name=name, schemas=schemas, required=required, parent_name=parent_name)
```

Comment:
I think you have to call `resolve_references` here in order to not miss any references in allOf from non-component schemas.